IBX-2752 Raptor CDP connector (#1756)
ezsystems/developer-documentation: docs/cdp/cdp_activation.md

> # CDP activation

Author: ```suggestion
---
description: Step by step activation procedure of Ibexa CDP.
---

# CDP activation
```

> # CDP activation  [[% include 'snippets/experience_badge.md' %]] [[% include 'snippets/commerce_badge.md' %]]

Reviewer: See my previous comment about badges.

## Configuration

> To configure Ibexa CDP you have to edit the `config/packages/ibexa_cdp.yaml` file:

Reviewer 2: ```suggestion
To configure Ibexa CDP, edit the `config/packages/ibexa_cdp.yaml` file:
```

```yaml
ibexa:
    system:
        default:
            cdp:
                account_number: 123456
                user_data_streaming:
                    stream_id: 00000000-00000000-00000000-00000000
                activations:
                    client_id: '%env(CDP_ACTIVATION_CLIENT_ID)%'
                    client_secret: '%env(CDP_ACTIVATION_CLIENT_SECRET)%'
                    segment_group_identifier: example_segment_group_identifier
```

> All configuration settings are described bellow.

Reviewer: A typo:
```suggestion
All configuration settings are described below.
```

You can follow them step by step to set up your Ibexa CDP.

### Segment group

First, create a segment group in the Back Office.
It will serve as a container for all segments data generated by Ibexa CDP.
Go to **Admin** -> **Segments** and select **Create**.
Fill in name and identifier for a segment group.
Choose wisely, as once connected to CDP Segment Group cannot be changed.

![Creating a new segment group](img/cdp_create_segment_group.png)

Next, add a segment group identifier to the configuration.

!!! caution "Ibexa CDP Segment Group"

>     Once you create the Segment Group in the Back Office and connect it to Ibexa CDP you cannot change it in any way, including edit its name.

Reviewer 2: ```suggestion
    After you create the Segment Group in the Back Office and connect it to Ibexa CDP, you cannot change it in any way, including edit its name.
```

## Account number

Now, fill in the account number.

> Log in to Ibexa CDP and select available accounts in the top right corner.

Reviewer 2: ```suggestion
Log in to Ibexa CDP and in the top right corner, select available accounts.
```

![List of available accounts](img/cdp_accounts.png)

> A pop-up window will appear with a list of all available accounts and their numbers.

Reviewer 2: ```suggestion
A pop-up window displays with a list of all available accounts and their numbers.
```

![Account number](img/cdp_account_number.png)

## User Data Streaming

You need to specify a source of the user data that Ibexa CDP will connect to.
To do so, go to **Data Manager** in **Tools** section and select **Create new dataflow**.
It will take you to a Dataflow Creator, where in five steps you will set up a data streaming.

### General Information

> In the **General Information** section specify dataflow name,

Reviewer 2: ```suggestion
In the **General Information** section, specify dataflow name,
```

choose **Stream File** as a source of user data and **CDP** as a destination,
where they will be sent for processing.

### Download

> In the **Download** section select **Stream file**. 

Reviewer 2: ```suggestion
In the **Download** section, select **Stream file**. 
```

Copy generated steam ID and paste it into the `config/packages/ibexa_cdp.yaml` file under `stream_id`.
It allows you to establish a datastream from the Streaming API into the Data Manager.

Next, you need to export your user data to the CDP.

> Got to your installation and use this command:

Reviewer 2: ```suggestion
Go to your installation and use this command:
```

```bash
php bin/console ibexa:cdp:stream-user-data --draft
```

There are two versions of this command `--draft/--no-draft`.
The first one is used to send the test user data to the Data Manager.

> If it passes a validation test in the **Activation** section, you will use the latter one to send a full version.

Reviewer 2: ```suggestion
If it passes a validation test in the **Activation** section, use the latter one to send a full version.
```

Next, go back to Ibexa CDP and select **Validate & download**.
If the file passes, you will see a confirmation message.

> Now, you can move on to the **File mapping** section.

Reviewer 2: ```suggestion
Now, you can go to the **File mapping** section.
```

Next, select **userid** from a **Schema columns section** on the right and map it to **id**.

![Map userid to id](img/cdp_userid_mapid.png)

### Activation

In this section you will test the dataflow with provided test user data.
If everything passes, go to your installation and export production data with this command:

```bash
php bin/console ibexa:cdp:stream-user-data --no-draft
```

Now you can run and activate the dataflow.

### Build new Audience/Segment

Go to **Audience Builder** and select **Build new audience**.
When naming the audience remember, you will need to find it in a drop-down list during activation.
There, you can choose conditions from `did`, `did not` or `have`.
The conditions `did` and `did not` allow you to use events like buy, visit or add to a basket from online tracking.
- `have` conditions are tied to personal characteristics and can be used to track the sum of all buys or top-visited categories.

In the Audience Builder you can also connect created audiences to the activations.

## Activation

Activation synchronises data from Ibexa CDP to the Ibexa DXP.

> When you specify a segment, you can Activate it on multiple communication channels, such as newsletters or commercials.

Reviewer 2: ```suggestion
When you specify a segment, you can activate it on multiple communication channels, such as newsletters or commercials.
```

You can configure multiple activations based on one user data streaming.

> First, select **Activations** from the menu bar and create a new **Ibexa** activation.

Reviewer 2: ```suggestion
First, from the menu bar, select **Activations**  and create a new **Ibexa** activation.
```

![Ibexa Information - Activation](img/cdp_activation_ibexa_info.png)

Finally, you can specify the audiences you wish to include.

!!! note "CDP requests"

    All CDP requests are logged in with `debug` severity.

## Add Client-side Tracking

The final step is setting up a tracking script.
It requires a head tracking script between the `<head></head>` tags on your website
and a main script after the head script, and cookie consent.
You can do it by following [tutorial in the documentation](https://support.raptorsmartadvisor.com/hc/en-us/articles/115000656909-Client-side-Tracking).

Now, you need to add a tracker to specific places in your website where you want to track users.
For example, add this tracker to the Landing Page template if you want to track user entrances.

```js
raptor.trackEvent('visit', ..., ...);
```
or buys:

```js
  //Parameters for Product 1
raptor.trackEvent('buy', ..., ...);
  //Parameters for Product 2
raptor.trackEvent('buy', ..., ...);
```

For tracing to be effective, you also need to send ID of a logged-in user in the same way.
Add the user ID information by using below script:

```js
raptor.push("setRuid","USER_ID_HERE")
```

> For more information on tracking events, see [the documentation]()https://support.raptorsmartadvisor.com/hc/en-us/articles/201912411-Tracking-Events.

Author: ```suggestion
For more information on tracking events, see [the documentation](https://support.raptorsmartadvisor.com/hc/en-us/articles/201912411-Tracking-Events).
```

> For more information on tracking events follow [the documentation]()https://support.raptorsmartadvisor.com/hc/en-us/articles/201912411-Tracking-Events.

Reviewer 2: ```suggestion
For more information on tracking events, see [the documentation]()https://support.raptorsmartadvisor.com/hc/en-us/articles/201912411-Tracking-Events.
```
